packages/camera/camera_windows/windows/camera_plugin.cpp

```diff
@@ -0,0 +1,590 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "camera_plugin.h"
+
+#include <flutter/flutter_view.h>
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar_windows.h>
+#include <flutter/standard_method_codec.h>
+#include <mfapi.h>
+#include <mfidl.h>
+#include <shlobj.h>
+#include <shobjidl.h>
+#include <windows.h>
+
+#include <cassert>
+#include <chrono>
+#include <memory>
+
+#include "com_heap_ptr.h"
+#include "device_info.h"
+#include "string_utils.h"
+
+namespace camera_windows {
+using flutter::EncodableList;
+using flutter::EncodableMap;
+using flutter::EncodableValue;
+
+namespace {
+
+// Channel events
+constexpr char kChannelName[] = "plugins.flutter.io/camera";
+
+constexpr char kAvailableCamerasMethod[] = "availableCameras";
+constexpr char kCreateMethod[] = "create";
+constexpr char kInitializeMethod[] = "initialize";
+constexpr char kTakePictureMethod[] = "takePicture";
+constexpr char kStartVideoRecordingMethod[] = "startVideoRecording";
+constexpr char kStopVideoRecordingMethod[] = "stopVideoRecording";
+constexpr char kPausePreview[] = "pausePreview";
+constexpr char kResumePreview[] = "resumePreview";
+constexpr char kDisposeMethod[] = "dispose";
+
+constexpr char kCameraNameKey[] = "cameraName";
+constexpr char kResolutionPresetKey[] = "resolutionPreset";
+constexpr char kEnableAudioKey[] = "enableAudio";
+
+constexpr char kCameraIdKey[] = "cameraId";
+constexpr char kMaxVideoDurationKey[] = "maxVideoDuration";
+
+constexpr char kResolutionPresetValueLow[] = "low";
+constexpr char kResolutionPresetValueMedium[] = "medium";
+constexpr char kResolutionPresetValueHigh[] = "high";
+constexpr char kResolutionPresetValueVeryHigh[] = "veryHigh";
+constexpr char kResolutionPresetValueUltraHigh[] = "ultraHigh";
+constexpr char kResolutionPresetValueMax[] = "max";
+
+const std::string kPictureCaptureExtension = "jpeg";
+const std::string kVideoCaptureExtension = "mp4";
+
+// Looks for |key| in |map|, returning the associated value if it is present, or
+// a nullptr if not.
+const EncodableValue* ValueOrNull(const EncodableMap& map, const char* key) {
+  auto it = map.find(EncodableValue(key));
+  if (it == map.end()) {
+    return nullptr;
+  }
+  return &(it->second);
+}
+
+// Looks for |key| in |map|, returning the associated int64 value if it is
+// present, or a nullptr if not.
+const int64_t* Int64OrNull(const EncodableMap& map, const char* key) {
+  auto value = ValueOrNull(map, key);
+  if (!value) {
+    return nullptr;
+  }
+
+  if (std::holds_alternative<int32_t>(*value)) {
+    return (int64_t*)std::get_if<int32_t>(value);
```

Comment:
as these were pointers, there was actually bug there. and as they were const pointers casting was not so straight forward either. as this function is not used so ofter, I just ended up copying the value instead of having pointer to it.